### PR TITLE
feat: add initial sources

### PR DESCRIPTION
**Pull request by utkuozdemir**
The initial implementation.

Depends on: https://github.com/siderolabs/omni/pull/449

### CODE REVIEW

File: .kres.yaml
Comment (utkuozdemir) [inline .kres.yaml]: This provisioner needs `talosctl`, as the provisioning code imported from Talos requires it to spawn more processes over it. The `talosctl` binary is searched in the working directory and in the PATH by this provisioner. When we use it as the base image, it is found in the current working directory.

File: api/specs/specs.proto
Comment (utkuozdemir) [inline api/specs/specs.proto:15]: These are the resources internal to this provider, living in the `cloud-provider:qemu` namespace of Omni's state.

File: cmd/omni-cloud-provider-qemu/main.go
Comment (utkuozdemir) [inline cmd/omni-cloud-provider-qemu/main.go:118]: This is the total pool of machines the provider provisions on startup. They go into an ipxe boot loop until they are assigned to a `MachineRequest`.

File: .kres.yaml
Comment (frezbo) [reply .kres.yaml]: i think we can just use a copy stage and copy over from the talosctl image

File: hack/release.toml
Comment (frezbo) [inline hack/release.toml]: weird, this shoudn't be changed

File: internal/provisioner/provisioner.go
Comment (frezbo) [inline internal/provisioner/provisioner.go]: probably we can pick latest always?

File: .kres.yaml
Comment (utkuozdemir) [reply .kres.yaml]: I look into that, but it looked like I can copy over from other images (in the same dockerfile), but not from an arbitrary image. Might be wrong though.

Also, does this have any downsides? There is nothing in the talosctl base image other than talosctl.

File: hack/release.toml
Comment (utkuozdemir) [reply hack/release.toml]: reverted that manually, they stay after rekres.

File: internal/provider/provider.go
Comment (Unix4ever) [inline internal/provider/provider.go:156]: I wonder if we should add a proper health endpoint.

**Review by Unix4ever — APPROVED** [review]

File: internal/provisioner/provisioner.go
Comment (utkuozdemir) [reply internal/provisioner/provisioner.go]: good point, done 👍

File: internal/provider/provider.go
Comment (utkuozdemir) [reply internal/provider/provider.go:156]: Yep, that's a good idea. Here the problem I work around is, when Omni gets restarted, the `state` I have over the client API gets broken, but the runtime doesn't become aware of it. I couldn't find a way to make it aware of it (using GRPC options etc.). Maybe there is one, but I couldn't find any.

So not sure if it's an Omni problem - maybe a COSI one?

**Comment by utkuozdemir** [comment]
Holding this until the Omni side gets merged

**Pull request merged**
